### ISSUE TITLE
Bug: RuleTester tests only the first output

### ISSUE THREAD

**Issue by daiyam** [typescript-eslint/typescript-eslint#12865]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.
- [x] If I used AI assistance to prepare this issue, I have reviewed and followed the [AI Contribution Policy](https://typescript-eslint.io/contributing/ai-policy/).

### Relevant Package

rule-tester

### Repro Code

```TypeScript
None at the moment.
```

### ESLint Config

```javascript

```

### tsconfig

```jsonc

```

### Expected Result

When passing `output` as a string to `RuleTester` for an invalid test, it should test the final output (if the rule produces multiple outputs).

### Actual Result

Currently, it only tests the first output and discards the other outputs.

### Additional Info

The issue is at:
https://github.com/typescript-eslint/typescript-eslint/blob/cd2e3d796b5e18cdaa4ea10fec0b87caaa8db026/packages/rule-tester/src/RuleTester.ts#L1399

The `result.outputs[0]` should be replaced with `result.outputs.at(-1)` so it can test the final output.
It should be replace at line 1391 and 1420.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/rule-tester`       | `8.70.0` |
